// natural-orderby/lib/utils/regex.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var RE_NUMBERS = /(^0x[\da-fA-F]+$|^([+-]?(?:\d+(?:\.\d*)?|\.\d+)(?:[eE][+-]?\d+)?(?!\.\d+)(?=\D|\s|$))|\d+)/g;
var RE_LEADING_OR_TRAILING_WHITESPACES = /^\s+|\s+$/g; // trim pre-post whitespace
var RE_WHITESPACES = /\s+/g; // normalize all whitespace to single ' ' character
var RE_INT_OR_FLOAT = /^[+-]?(?:\d+(?:\.\d*)?|\.\d+)(?:[eE][+-]?\d+)?$/; // identify integers and floats
var RE_DATE = /(^([\w ]+,?[\w ]+)?[\w ]+,?[\w ]+\d+:\d+(:\d+)?[\w ]?|^\d{1,4}[/-]\d{1,4}[/-]\d{1,4}|^\w+, \w+ \d+, \d{4})/; // identify date strings
var RE_LEADING_ZERO = /^0+[1-9]{1}[0-9]*$/;
var RE_UNICODE_CHARACTERS = /[^\x00-\x80]/;

exports.RE_NUMBERS = RE_NUMBERS;
exports.RE_LEADING_OR_TRAILING_WHITESPACES = RE_LEADING_OR_TRAILING_WHITESPACES;
exports.RE_WHITESPACES = RE_WHITESPACES;
exports.RE_INT_OR_FLOAT = RE_INT_OR_FLOAT;
exports.RE_DATE = RE_DATE;
exports.RE_LEADING_ZERO = RE_LEADING_ZERO;
exports.RE_UNICODE_CHARACTERS = RE_UNICODE_CHARACTERS;
